lib/bundler/resolver/spec_group.rb

```diff
@@ -58,18 +64,26 @@ def ==(other)
         return unless other.is_a?(SpecGroup)
         name == other.name &&
           version == other.version &&
+          sorted_activated_platforms == other.sorted_activated_platforms &&
           source == other.source
       end
 
       def eql?(other)
         return unless other.is_a?(SpecGroup)
         name.eql?(other.name) &&
           version.eql?(other.version) &&
+          sorted_activated_platforms.eql?(other.sorted_activated_platforms) &&
```

Comment:
This is one of the most important changes in this pull request.
Without this change, `SpecGroup`s that have different activated platforms are the same `SpecGroup`.
With this change, `SpecGroup`s that have different activated platforms are the different `SpecGroup`s.

By this change, we can choose a gem that is conflicted with `x64-mingw32` version but is available with `ruby` version. See added examples into `spec/resolver/platform_spec.rb` for details.